scippy/tests/test_variable.py

```diff
@@ -0,0 +1,225 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2019 Scipp contributors (https://github.com/scipp)
+# @file
+# @author Simon Heybrock
+import pytest
+
+import scippy as sp
+from scippy import Dim
+import numpy as np
+
+
+def make_variables():
+    data = np.arange(1, 4, dtype=float)
+    a = sp.Variable([sp.Dim.X], data)
+    b = sp.Variable([sp.Dim.X], data)
+    a_slice = a[sp.Dim.X, :]
+    b_slice = b[sp.Dim.X, :]
+    return a, b, a_slice, b_slice, data
+
+
+def test_create_default():
+    var = sp.Variable()
+    assert var.dims == sp.Dimensions()
+    assert var.dtype == sp.dtype.double
+    assert var.unit == sp.units.dimensionless
+    assert var.value == 0.0
+
+
+def test_create_default_dtype():
+    var = sp.Variable([sp.Dim.X], [4])
+    assert var.dtype == sp.dtype.double
+
+
+def test_create_with_dtype():
+    var = sp.Variable(labels=[Dim.X], shape=[2], dtype=sp.dtype.float)
+    assert var.dtype == sp.dtype.float
+
+
+def test_create_with_numpy_dtype():
+    var = sp.Variable(labels=[Dim.X], shape=[2], dtype=np.dtype(np.float32))
+    assert var.dtype == sp.dtype.float
+
+
+def test_create_with_variances():
+    assert not sp.Variable(labels=[Dim.X], shape=[2]).has_variances
+    assert not sp.Variable(labels=[Dim.X], shape=[
+                           2], variances=False).has_variances
+    assert sp.Variable(labels=[Dim.X], shape=[2], variances=True).has_variances
```

Comment:
Not that it makes much sense, but
what happens to the `variances=True` syntax if you are creating a `Variable` of type bool with a single value and you add a variance that is either `True` or `False`?

Reply:
Unless you specify the `values` argument you should always match the other overload, so this cannot happen, I think?